src/test/java/seedu/address/ui/CommandBoxTest.java

```diff
@@ -68,6 +68,18 @@ public void commandBox_handleKeyPush() {
         assertEquals(defaultStyleOfCommandBox, commandBoxHandle.getStyleClass());
     }
 
+    @Test
+    public void commandBox_correctHistory() {
+        String viewCommand1 = "client view 1";
+        String viewCommand2 = "client view 2";
+        commandBoxHandle.run(viewCommand1);
+        commandBoxHandle.run(viewCommand2);
+        guiRobot.pauseForHuman();
+        guiRobot.press(KeyCode.UP);
+        String prevCommand = commandBoxHandle.getInput();
+        assertEquals(viewCommand2, prevCommand);
+    }
+
```

Comment:
Should add more tests that do invalid commands then check if the history is correct, and also test the behavior with no history, and the behavior when there's no previous history, and lastly the behavior when there's no next history.